src/opencl_common.c

```diff
@@ -663,10 +687,30 @@ static void add_device_to_list(int sequential_id)
 	requested_devices[get_number_of_requested_devices()] = sequential_id;
 }
 
-static void add_device_type(cl_ulong device_type)
+/* Used below (inside add_device_type routine) to sort devices */
+typedef struct {
+	int index;
+	cl_device_id ID;
+	unsigned int value;
+} speed_sort_t;
+
+/* Used below (inside add_device_type routine) to sort devices */
+static int comparator(const void *p1, const void *p2)
+{
+	const speed_sort_t *c1 = (const speed_sort_t *)p1;
+	const speed_sort_t *c2 = (const speed_sort_t *)p2;
+	int diff = (int)c2->value - (int)c1->value;
+	if (diff)
+		return diff;
+	return c1->index - c2->index;
+}
```

Comment:
This will ensure that devices with same speed will end up sorted by device number, lowest first, right? I like it.

Reply:
Yep. Well, this is how JtR proper works, I just keep it.